packages/vite/src/node/utils.ts

```diff
@@ -642,3 +642,46 @@ export function parseRequest(id: string): Record<string, string> | null {
   }
   return Object.fromEntries(new URLSearchParams(search.slice(1)))
 }
+
+// reg without the 'g' option, only matches the first match
+const multilineCommentsOnceRE = /\/\*(.|[\r\n])*?\*\//m
+const singlelineCommentsOnceRE = /\/\/.*/
+
+// find index ignore comment
+export function findIndex(code: string, i: number, endChar: string): number {
+  const findStart = i
+  const endIndex = code.indexOf(endChar, findStart)
+  const subCode = code.substring(findStart)
+
+  const matchedSingleline = subCode.match(singlelineCommentsOnceRE)
+  const matchedMultiline = subCode.match(multilineCommentsOnceRE)
+
+  if (!matchedSingleline && !matchedMultiline) {
+    return endIndex
+  }
+
+  let matched: RegExpMatchArray
+  if (!matchedSingleline || !matchedMultiline) {
+    matched = matchedSingleline ?? matchedMultiline!
+  } else {
+    if (matchedSingleline.index! > matchedMultiline.index!) {
+      matched = matchedMultiline
+    } else {
+      matched = matchedSingleline
+    }
+  }
+
+  const str = matched[0]
+  const index = matched.index
+  if (!index) {
+    return endIndex
+  }
+
+  const commentStart = findStart + index
+  const commentEnd = commentStart + str.length
+  if (endIndex > commentStart && endIndex < commentEnd) {
+    return findIndex(code, commentEnd, endChar)
+  } else {
+    return endIndex
+  }
```

Comment:
Maybe this isn't the right place to fix this, so we may want to do another PR before this one. @ygj6 could there be a bug here? Imagine trying to find `)` here
```
  // comment
  // comment )
```
Or here
```
  // comment
  /* comment ) */
```
The first comment will match, and the `endIndex` will be bigger than `commentEnd` so it doesn't enter the `findIndex` branch.
Should we only return the `endIndex` if it is before the comment?
```suggestion
  if (endIndex < commentStart) {
    return endIndex
  }
  else {
    return findIndex(code, commentEnd, endChar)
  }
```
Another option to simplify this function may be to replace the comments with empty space so indexes remain stable and then do the search.
```ts
// reg with the 'g' option, matches all
const multilineCommentsRE = /\/\*(.|[\r\n])*?\*\//gm
const singlelineCommentsRE = /\/\/.*/g

const blankReplacer = (match: string) => ' '.repeat(match.length)

export function findIndex(code: string, i: number, endChar: string): number {
  return code
    .replace(multilineCommentsRE,blankReplacer)
    .replace(singlelineCommentsRE,blankReplacer)
    .indexOf(endChar, findStart)
}
```

Reply:
greet!

Reply:
but `code` had long char, replacer every time may had high performance overhead

Reply:
I could like to had this funtion(find index ignore comment) and I try to fix the funtion.

Reply:
@patak-dev Very good point! I personally prefer the second option because it's more stable.

> but code had long char, replacer every time may had high performance overhead

@poyoho I think that when we try to `findIndex` without commenting, we'd better off trying to find in less code after pruning or slicing.

Step back and I see that there is already `noCommentCode` here, and we don't need to do it again in `getWorkerType`.
https://github.com/vitejs/vite/blob/87c9e3d08e10b31f52245ed7efd8fe8708d8bc4b/packages/vite/src/node/plugins/workerImportMetaUrl.ts#L38-L40

Reply:
yep, I thought about it. I can't to fix it. because we need fine every comment does it exist `endChar`.

this greet point. @patak-dev

Reply:
Would you expand on the issue you are seeing with this approach?